sdk/src/main/scala/io/horizen/forge/AbstractForger.scala

```diff
@@ -46,6 +46,7 @@ abstract class AbstractForger[
   type View = CurrentView[HIS, MS, VL, MP]
 
   private val restApiTimeoutDuration: FiniteDuration = settings.sparkzSettings.restApi.timeout
+  private var slotsDurationInSeconds = ConsensusParamsUtil.getConsensusSecondsInSlotsPerEpoch(params.sidechainGenesisBlockTimestamp, timeProvider.time() / 1000) * 1000
```

Comment:
looks to me not 100% safe to initialize this here (the time the class is instantiated and the time startTimer is called for the first time could differ?)
we may initialize this to null and in startTimer add something like this:
if (slotsDurationInSeconds == null) {
   slotsDurationInSeconds  =  ConsensusParamsUtil.getConsensusSecondsInSlotsPerEpoch(params.sidechainGenesisBlockTimestamp, timeProvider.time() / 1000) * 1000
}

Reply:
Done